syntax/javascript.vim

```diff
@@ -337,7 +337,7 @@ if version >= 508 || !exists("did_javascript_syn_inits")
   HiLink jsSwitchColon          Noise
   HiLink jsClassMethodType      Type
   HiLink jsObjectMethodType     Type
-  HiLink jsClassDefinition      jsFuncName
+  HiLink jsClassDefinition      Constant
```

Comment:
Basically the idea being here - we aren't trying to be opinionated about things, just provide syntactically accurate matching (or at least as close as possible, given the limitations of trying to use a regexes to parse a language design for a lexical parser)